compiler/qsc_frontend/src/closure.rs

```diff
@@ -63,7 +61,7 @@ impl Visitor<'_> for VarFinder {
 }
 
 struct VarReplacer<'a> {
-    substitutions: &'a HashMap<NodeId, NodeId>,
+    substitutions: &'a FxHashMap<NodeId, NodeId>,
```

Comment:
Same question here about `IndexMap<NodeId, T>` vs. a hash map.

Basically I have the same question everywhere a `FxHashSet<FooId>` or `FxHashMap<FooId, T>` occurs in the PR... I'll refrain from commenting on every single one :)

Reply:
Great question @minestarks - I had hoped to do a follow up and see where we can use `IndexMap`, but there is a density issue. If we don't have a dense map, then `IndexMap` could allocate a lot of resources for very few entries. For example, if the `IndexMap` is given `NodeId` `22358`, then we just allocated a buffer with 22k+ slots for a single value. This PR is just a starting point.

Reply:
I did have that similar concern. But then my question becomes, how are we so sure that the _other_ `IndexMap<NodeId>`s in the code won't start at `22358`? I can see a few spots where it's almost guaranteed the map will contain most of the nodes in the program, but I'm not so sure about other spots. 

That being said, I agree this is out of scope for this PR, and I'm happy to hear you plan to keep digging!